lib/buffer.js

```diff
@@ -351,7 +364,7 @@ Buffer.concat = function(list, length) {
     length = length >>> 0;
   }
 
-  var buffer = Buffer.allocUnsafe(length);
+  var buffer = allocate(length);
```

Comment:
Yeah, just part of "performance improvement" by removing unnecessary indirections where safe to do so, but irrelevant to main optimizations.